client/js/upload-handler/upload.handler.controller.js

```diff
@@ -177,7 +177,8 @@ qq.UploadHandlerController = function(o, namespace) {
                 chunkData = handler._getChunkData(id, chunkIdx),
                 fileState = handler._getFileState(id),
                 resuming = fileState.attemptingResume,
-                inProgressChunks = fileState.chunking.inProgress || [];
+                inProgressChunks = fileState.chunking.inProgress || [],
+                minSuccessfulUploadResponseLength = 20;
```

Comment:
Why is this `20`?

Reply:
@navitronic So, this is a bit of a hack job....

FM responds with `{"success": "true"}` when a chunk is successfully uploaded, but the FM is still waiting on other chunks.  FM responds with `{"success": "true", "file_name": "ewkfewjfehwfhefbjw.pdf"...}` when it's the final chunk.  This is our way of working out whether it is the final chunk.  If there's a better way to do this, please let us know....

Reply:
Basically, if it's the final chunk - as per the FM - (but perhaps not the last chunk to be received by the client) we want to *not* clear it from the history books, as it is needed in order to finalise the upload (trigger a commit, etc)